--- nagios-plugins-1.4.14/contrib/check_rbl.c~	2009-09-17 11:33:40.000000000 +0300
+++ nagios-plugins-1.4.14/contrib/check_rbl.c	2009-09-17 11:33:41.629634485 +0300
@@ -39,6 +39,10 @@
 char *dns_server=NULL;
 char *rbl_name=NULL;
 int verbose=FALSE;
+#ifndef LOCAL_TIMEOUT_ALARM_HANDLER
+unsigned int socket_timeout = DEFAULT_SOCKET_TIMEOUT;
+unsigned int socket_timeout_state = STATE_CRITICAL;
+#endif
 
 int main(int argc, char **argv){
 	char input_buffer[MAX_INPUT_BUFFER];
