--- nagios-plugins-2.1.4.org/commands/check_http.cfg	2014-03-07 13:13:31.000000000 +0100
+++ nagios-plugins-2.1.4/commands/check_http.cfg	2017-08-20 21:40:54.871308710 +0200
@@ -1,17 +1,17 @@
 # 'check_http' command definition
 define command {
 	command_name    check_http
-	command_line    $USER1$/check_http -I $HOSTADDRESS$ -f warning $ARG1$
+	command_line    $USER1$/check_http -I $HOSTADDRESS$ -H $HOSTNAME$ -f warning $ARG1$
 }
 
 define command {
 	command_name    check_https
-	command_line    $USER1$/check_http -I $HOSTADDRESS$ -f warning -S $ARG1$
+	command_line    $USER1$/check_http -I $HOSTADDRESS$ -H $HOSTNAME$ -f warning -S --sni $ARG1$
 }
 
 define command {
 	command_name    check_https.crt
-	command_line    $USER1$/check_http -I $HOSTADDRESS$ -S -C 14 $ARG1$
+	command_line    $USER1$/check_http -I $HOSTADDRESS$ -H $HOSTNAME$ -S -C 14 --sni $ARG1$
 }
 
 define service {
