--- plugins-scripts/subst.in~	2002-06-06 07:16:56.000000000 +0300
+++ plugins-scripts/subst.in	2004-12-08 14:58:28.000000000 +0200
@@ -44,9 +44,6 @@
 /^#! ?\/.*\/[a-z]{0,2}awk/ {sub(/^#! ?\/.*\/[a-z]{0,2}awk/,"#! @AWK@");}
 /^#! ?\/.*\/sh/ {sub(/^#! ?\/.*\/sh/,"#! @SHELL@");}
 
-# add to libexecdir to INC for perl utils.pm
-/^use/ { if (/lib/) { if (/utils.pm/ ) {sub(/utils.pm/,led() )} } }
-
 
 # Trusted path mechanism (deprecated)
 
@@ -68,6 +65,9 @@
 	sub(c,which(c,path));
 }
 
+# add to libexecdir to INC for perl utils.pm
+/^use/ { if (/lib/) { if (/utils.pm/ ) {sub(/utils.pm/,led() )} } }
+
 {
 	print;
 }
