diff -urNp -x '*.orig' nagios-plugins-2.3.3.org/plugins-root/Makefile.am nagios-plugins-2.3.3/plugins-root/Makefile.am
--- nagios-plugins-2.3.3.org/plugins-root/Makefile.am	2020-02-20 20:14:52.000000000 +0100
+++ nagios-plugins-2.3.3/plugins-root/Makefile.am	2021-04-19 12:00:59.891325797 +0200
@@ -46,10 +46,6 @@ INSTALL_SUID = \
 	p=$$f; \
 	echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(libexecdir)/$$p"; \
 	$(INSTALL_PROGRAM) $$p $(DESTDIR)$(libexecdir)/$$p; \
-	echo " chown root $(DESTDIR)$(libexecdir)/$$p"; \
-	chown root $(DESTDIR)$(libexecdir)/$$p; \
-	echo " chmod $(setuid_root_mode) $(DESTDIR)$(libexecdir)/$$p"; \
-	chmod $(setuid_root_mode) $(DESTDIR)$(libexecdir)/$$p; \
 	done
 
 INSTALL_NOSUID = \
