--- nagios-plugins-1.4.13/contrib/check_snmp_process_monitor.pl~	2008-10-11 14:43:23.000000000 +0300
+++ nagios-plugins-1.4.13/contrib/check_snmp_process_monitor.pl	2008-10-11 14:47:46.319041094 +0300
@@ -115,7 +115,7 @@
 -e, --command=COMMAND NAME (ps -e style)
    what command should be monitored?
 -r, --regex=Perl RE
-   use a perl regular expression to find your process
+   a Perl regular expression to find your process
 -w, --warning=INTEGER[,INTEGER]
    minimum and maximum number of processes before a warning is issued (Default 1,-1)
 -c, --critical=INTEGER[,INTEGER]
