--- ./configure.in~	2005-03-08 02:42:40.562147519 +0200
+++ ./configure.in	2005-03-08 02:54:45.648208868 +0200
@@ -843,6 +843,7 @@
 	if test -n "$ac_cv_ping_packets_first"
 	then
 		ac_cv_ping_packets_first=yes
+		ac_cv_ping_has_timeout=yes
 	fi
 
 elif [[ "z$ac_cv_uname_s" = "zUnixWare" ]] && \
--- ./configure.in~	2005-03-23 20:54:25.000000000 +0200
+++ ./configure.in	2005-03-23 21:09:57.000000000 +0200
@@ -575,11 +575,8 @@
 AC_DEFINE_UNQUOTED(SOCKET_SIZE_TYPE, $ac_cv_socket_size_type ,
 	[Define type of socket size])
 
-if test -f "/proc/loadavg"
-then
-  AC_DEFINE(HAVE_PROC_LOADAVG,1,[Define if /proc/loadavg or similar exists])
-  AC_DEFINE_UNQUOTED(PROC_LOADAVG,"/proc/loadavg",[Location of /proc/loadavg])
-fi
+AC_DEFINE(HAVE_PROC_LOADAVG,1,[Define if /proc/loadavg or similar exists])
+AC_DEFINE_UNQUOTED(PROC_LOADAVG,"/proc/loadavg",[Location of /proc/loadavg])
 
 dnl #### Process table test
 
--- ./configure.in~	2005-03-23 21:34:43.000000000 +0200
+++ ./configure.in	2005-03-23 21:35:08.000000000 +0200
@@ -583,10 +583,23 @@
 AC_PATH_PROG(PATH_TO_PS,ps)
 
 AC_MSG_CHECKING(for ps syntax)
+
+AC_ARG_WITH(ps_command,--with-ps-command=<syntax> Verbatim command to execute for ps in check_procs,PS_COMMAND=$withval)
+AC_ARG_WITH(ps_format,--with-ps-format=<format> Format string for scanning ps output in check_procs,PS_FORMAT=$withval)
+AC_ARG_WITH(ps_cols,--with-cols=<number> Number of columns in ps command,PS_COLS=$withval)
+AC_ARG_WITH(ps_varlist,--with-ps-varlist=<list> Variable list for sscanf of 'ps' output,PS_VARLIST=$withval)
+
+if test -n "$PS_COMMAND" && test -n "$PS_FORMAT" && test -n "$PS_COLS" && test -n "$PS_VARLIST"; then 
+	ac_cv_ps_varlist="$PS_VARLIST"
+	ac_cv_ps_command="$PS_COMMAND"
+	ac_cv_ps_format="$PS_FORMAT"
+	ac_cv_ps_cols="$PS_COLS"
+	AC_MSG_RESULT([(command-line) $ac_cv_ps_command])
+
 dnl For OpenBSD 3.2 & 3.3. Must come before ps -weo
 dnl Should also work for FreeBSD 5.2.1 and 5.3
 dnl  STAT UCOMM              VSZ   RSS USER      PPID COMMAND
-if ps -axwo 'stat comm vsz rss user uid ppid args' 2>/dev/null | \
+elif ps -axwo 'stat comm vsz rss user uid ppid args' 2>/dev/null | \
 	egrep -i ["^ *STAT +[UCOMAND]+ +VSZ +RSS +USER +UID +PPID +COMMAND"] > /dev/null
 then
 	ac_cv_ps_varlist="[procstat,&procuid,&procppid,&procvsz,&procrss,&procpcpu,procprog,&pos]"
--- ./configure.in~	2005-02-08 16:25:52.000000000 +0200
+++ ./configure.in	2005-03-23 21:50:10.000000000 +0200
@@ -1426,20 +1424,26 @@
 		[Conversion factor to MB])
 fi
 
+AC_ARG_WITH(proc-meminfo,--with-proc-meminfo=<path> path to /proc/meminfo if name changes,ac_cv_proc_meminfo=$withval)
 dnl dunno why this does not work below - use hack (kbd)
 dnl fine on linux, broken on solaris
 dnl if /bin/test -e "/proc/meminfo"
 AC_MSG_CHECKING([for /proc/meminfo])
-if [cat /proc/meminfo > /dev/null 2>&1]
-then
+if test -n "$ac_cv_proc_meminfo"; then
+	AC_MSG_RESULT([(command line) $ac_cv_proc_meminfo])
+elif [cat /proc/meminfo > /dev/null 2>&1]; then
 	AC_MSG_RESULT([found /proc/meminfo])
-	AC_DEFINE(HAVE_PROC_MEMINFO,1,[Define if we have /proc/meminfo])
-	AC_DEFINE_UNQUOTED(PROC_MEMINFO,"/proc/meminfo",[path to /proc/meminfo if name changes])
-	EXTRAS="$EXTRAS check_swap"
+	ac_cv_proc_meminfo="/proc/meminfo"
 else
 	AC_MSG_RESULT([no])
 fi
 
+if test -n "$ac_cv_proc_meminfo"; then
+	AC_DEFINE(HAVE_PROC_MEMINFO,1,[Define if we have /proc/meminfo])
+	AC_DEFINE_UNQUOTED(PROC_MEMINFO,"$ac_cv_proc_meminfo",[path to /proc/meminfo if name changes])
+	EXTRAS="$EXTRAS check_swap"
+fi
+
 AC_PATH_PROG(PATH_TO_DIG,dig)
 AC_DEFINE_UNQUOTED(PATH_TO_DIG,"$PATH_TO_DIG",[Path to dig command, if present])
 if test -n "$PATH_TO_DIG"; then
