--- nagios-plugins-1.4.15/plugins/check_hpjd.c~	2010-09-14 16:49:38.000000000 +0300
+++ nagios-plugins-1.4.15/plugins/check_hpjd.c	2010-09-14 16:52:30.060156750 +0300
@@ -70,6 +70,7 @@
 
 char *community = NULL;
 char *address = NULL;
+int nopaperout = 0;
 
 int
 main (int argc, char **argv)
@@ -243,7 +244,9 @@
 			strcpy (errmsg, _("Paper Jam"));
 		}
 		else if (paper_out) {
-			result = STATE_WARNING;
+			if (!nopaperout) {
+				result = STATE_WARNING;
+			}
 			strcpy (errmsg, _("Out of Paper"));
 		}
 		else if (line_status == OFFLINE) {
@@ -318,6 +321,7 @@
 /*  		{"critical",       required_argument,0,'c'}, */
 /*  		{"warning",        required_argument,0,'w'}, */
 /*  		{"port",           required_argument,0,'P'}, */
+		{"nopaperout", no_argument, 0, 'n'},
 		{"version", no_argument, 0, 'V'},
 		{"help", no_argument, 0, 'h'},
 		{0, 0, 0, 0}
@@ -328,7 +332,7 @@
 
 	
 	while (1) {
-		c = getopt_long (argc, argv, "+hVH:C:", longopts, &option);
+		c = getopt_long (argc, argv, "+hnVH:C:", longopts, &option);
 
 		if (c == -1 || c == EOF || c == 1)
 			break;
@@ -345,6 +349,9 @@
 		case 'C':									/* community */
 			community = strscpy (community, optarg);
 			break;
+		case 'n':									/* nopaperout */
+			nopaperout = 1;
+			break;
 		case 'V':									/* version */
 			print_revision (progname, revision);
 			exit (STATE_OK);
@@ -404,7 +411,9 @@
 
 	printf (" %s\n", "-C, --community=STRING");
 	printf ("    %s", _("The SNMP community name "));
+	printf (" %s\n", "-n, --nopaperout");
 	printf (_("(default=%s)"), DEFAULT_COMMUNITY);
+	printf ("    %s\n", "No return WARNING state on \"Out of paper\"");
 	printf ("\n");
 
 #ifdef NP_EXTRA_OPTS
@@ -421,5 +421,5 @@
 print_usage (void)
 {
   printf ("%s\n", _("Usage:"));
-	printf ("%s -H host [-C community]\n", progname);
+  printf ("%s -H host [-C community] [-n]\n", progname);
 }
