--- nagios-plugins-1.4.13/plugins-scripts/check_mailq.pl~	2006-07-05 16:45:57.000000000 +0300
+++ nagios-plugins-1.4.13/plugins-scripts/check_mailq.pl	2008-10-11 14:52:47.407968164 +0300
@@ -32,7 +32,7 @@
 use vars qw($opt_V $opt_h $opt_v $verbose $PROGNAME $opt_w $opt_c $opt_t
 					$opt_M $mailq $status $state $msg $msg_q $msg_p $opt_W $opt_C $mailq @lines
 					%srcdomains %dstdomains);
-use lib  utils.pm;
+use lib utils.pm;
 use utils qw(%ERRORS &print_revision &support &usage );
 
 
