diff -urNp -x '*.orig' nagios-plugins-2.3.3.org/plugins/check_hpjd.c nagios-plugins-2.3.3/plugins/check_hpjd.c
--- nagios-plugins-2.3.3.org/plugins/check_hpjd.c	2019-12-04 22:53:08.000000000 +0100
+++ nagios-plugins-2.3.3/plugins/check_hpjd.c	2021-04-19 12:01:52.206215349 +0200
@@ -41,7 +41,7 @@ const char *email = "devel@nagios-plugin
 #define DEFAULT_COMMUNITY "public"
 #define DEFAULT_PORT "161"
 
-const char *option_summary = "-H host [-C community]\n";
+const char *option_summary = "-H host [-C community] [-n]\n";
 
 #define HPJD_LINE_STATUS           ".1.3.6.1.4.1.11.2.3.9.1.1.2.1"
 #define HPJD_PAPER_STATUS          ".1.3.6.1.4.1.11.2.3.9.1.1.2.2"
@@ -68,6 +68,7 @@ char *community = NULL;
 char *address = NULL;
 int port = 0;
 char flawcorrection =0; // There are devices which report offline when that is not the case. Correct that.
+int nopaperout = 0;
 
 int
 main (int argc, char **argv)
@@ -251,7 +252,9 @@ main (int argc, char **argv)
 			strcpy (errmsg, _("Paper Jam"));
 		}
 		else if (paper_out) {
-			result = STATE_WARNING;
+			if (!nopaperout) {
+				result = STATE_WARNING;
+			}
 			strcpy (errmsg, _("Out of Paper"));
 		}
 		else if (line_status == OFFLINE) {
@@ -326,6 +329,7 @@ process_arguments (int argc, char **argv
 /*  		{"critical",       required_argument,0,'c'}, */
 /*  		{"warning",        required_argument,0,'w'}, */
   		{"port", required_argument,0,'p'}, 
+		{"nopaperout", no_argument, 0, 'n'},
 		{"flawcorrection", no_argument, 0, 'N'},
 		{"version", no_argument, 0, 'V'},
 		{"help", no_argument, 0, 'h'},
@@ -337,7 +341,7 @@ process_arguments (int argc, char **argv
 
 
 	while (1) {
-		c = getopt_long (argc, argv, "+hVNH:C:p:", longopts, &option);
+		c = getopt_long (argc, argv, "+hnNVH:C:p:", longopts, &option);
 
 		if (c == -1 || c == EOF || c == 1)
 			break;
@@ -360,6 +364,9 @@ process_arguments (int argc, char **argv
 			else
 				port = atoi(optarg);
 			break;
+		case 'n':									/* nopaperout */
+			nopaperout = 1;
+			break;
                 case 'N':                                                                       /* flaw correction */
                         flawcorrection=1;
                         break;
@@ -429,7 +436,9 @@ print_help (void)
 
 	printf (" %s\n", "-C, --community=STRING");
 	printf ("    %s", _("The SNMP community name "));
+	printf (" %s\n", "-n, --nopaperout");
 	printf (_("(default=%s)"), DEFAULT_COMMUNITY);
+	printf ("    %s\n", "No return WARNING state on \"Out of paper\"");
 	printf ("\n");
 	printf (" %s\n", "-p, --port=STRING");
 	printf ("    %s", _("Specify the port to check "));
@@ -449,5 +458,5 @@ void
 print_usage (void)
 {
   printf ("%s\n", _("Usage:"));
-	printf ("%s -H host [-C community] [-p port][-N]\n", progname);
+	printf ("%s -H host [-C community] [-p port] [-N] [-n]\n", progname);
 }
