--- ./configure.in~	2005-03-08 02:42:40.562147519 +0200
+++ ./configure.in	2005-03-08 02:54:45.648208868 +0200
@@ -843,6 +843,7 @@
 	if test -n "$ac_cv_ping_packets_first"
 	then
 		ac_cv_ping_packets_first=yes
+		ac_cv_ping_has_timeout=yes
 	fi
 
 elif [[ "z$ac_cv_uname_s" = "zUnixWare" ]] && \
--- ./configure.in~	2005-03-23 20:54:25.000000000 +0200
+++ ./configure.in	2005-03-23 21:09:57.000000000 +0200
@@ -575,11 +575,8 @@
 AC_DEFINE_UNQUOTED(SOCKET_SIZE_TYPE, $ac_cv_socket_size_type ,
 	[Define type of socket size])
 
-if test -f "/proc/loadavg"
-then
-  AC_DEFINE(HAVE_PROC_LOADAVG,1,[Define if /proc/loadavg or similar exists])
-  AC_DEFINE_UNQUOTED(PROC_LOADAVG,"/proc/loadavg",[Location of /proc/loadavg])
-fi
+AC_DEFINE(HAVE_PROC_LOADAVG,1,[Define if /proc/loadavg or similar exists])
+AC_DEFINE_UNQUOTED(PROC_LOADAVG,"/proc/loadavg",[Location of /proc/loadavg])
 
 dnl #### Process table test
 
--- ./configure.in~	2005-03-23 21:34:43.000000000 +0200
+++ ./configure.in	2005-03-23 21:35:08.000000000 +0200
@@ -583,10 +583,23 @@
 AC_PATH_PROG(PATH_TO_PS,ps)
 
 AC_MSG_CHECKING(for ps syntax)
+
+AC_ARG_WITH(ps_command,--with-ps-command=<syntax> Verbatim command to execute for ps in check_procs,PS_COMMAND=$withval)
+AC_ARG_WITH(ps_format,--with-ps-format=<format> Format string for scanning ps output in check_procs,PS_FORMAT=$withval)
+AC_ARG_WITH(ps_cols,--with-cols=<number> Number of columns in ps command,PS_COLS=$withval)
+AC_ARG_WITH(ps_varlist,--with-ps-varlist=<list> Variable list for sscanf of 'ps' output,PS_VARLIST=$withval)
+
+if test -n "$PS_COMMAND" && test -n "$PS_FORMAT" && test -n "$PS_COLS" && test -n "$PS_VARLIST"; then 
+	ac_cv_ps_varlist="$PS_VARLIST"
+	ac_cv_ps_command="$PS_COMMAND"
+	ac_cv_ps_format="$PS_FORMAT"
+	ac_cv_ps_cols="$PS_COLS"
+	AC_MSG_RESULT([(command-line) $ac_cv_ps_command])
+
 dnl For OpenBSD 3.2 & 3.3. Must come before ps -weo
 dnl Should also work for FreeBSD 5.2.1 and 5.3
 dnl  STAT UCOMM              VSZ   RSS USER      PPID COMMAND
-if ps -axwo 'stat comm vsz rss user uid ppid args' 2>/dev/null | \
+elif ps -axwo 'stat comm vsz rss user uid ppid args' 2>/dev/null | \
 	egrep -i ["^ *STAT +[UCOMAND]+ +VSZ +RSS +USER +UID +PPID +COMMAND"] > /dev/null
 then
 	ac_cv_ps_varlist="[procstat,&procuid,&procppid,&procvsz,&procrss,&procpcpu,procprog,&pos]"
--- ./configure.in~	2005-02-08 16:25:52.000000000 +0200
+++ ./configure.in	2005-03-23 21:50:10.000000000 +0200
@@ -1426,20 +1424,26 @@
 		[Conversion factor to MB])
 fi
 
+AC_ARG_WITH(proc-meminfo,--with-proc-meminfo=<path> path to /proc/meminfo if name changes,ac_cv_proc_meminfo=$withval)
 dnl dunno why this does not work below - use hack (kbd)
 dnl fine on linux, broken on solaris
 dnl if /bin/test -e "/proc/meminfo"
 AC_MSG_CHECKING([for /proc/meminfo])
-if [cat /proc/meminfo > /dev/null 2>&1]
-then
+if test -n "$ac_cv_proc_meminfo"; then
+	AC_MSG_RESULT([(command line) $ac_cv_proc_meminfo])
+elif [cat /proc/meminfo > /dev/null 2>&1]; then
 	AC_MSG_RESULT([found /proc/meminfo])
-	AC_DEFINE(HAVE_PROC_MEMINFO,1,[Define if we have /proc/meminfo])
-	AC_DEFINE_UNQUOTED(PROC_MEMINFO,"/proc/meminfo",[path to /proc/meminfo if name changes])
-	EXTRAS="$EXTRAS check_swap"
+	ac_cv_proc_meminfo="/proc/meminfo"
 else
 	AC_MSG_RESULT([no])
 fi
 
+if test -n "$ac_cv_proc_meminfo"; then
+	AC_DEFINE(HAVE_PROC_MEMINFO,1,[Define if we have /proc/meminfo])
+	AC_DEFINE_UNQUOTED(PROC_MEMINFO,"$ac_cv_proc_meminfo",[path to /proc/meminfo if name changes])
+	EXTRAS="$EXTRAS check_swap"
+fi
+
 AC_PATH_PROG(PATH_TO_DIG,dig)
 AC_DEFINE_UNQUOTED(PATH_TO_DIG,"$PATH_TO_DIG",[Path to dig command, if present])
 if test -n "$PATH_TO_DIG"; then
--- ./configure.in~	2005-02-08 16:25:52.000000000 +0200
+++ ./configure.in	2005-03-27 22:51:34.000000000 +0300
@@ -1089,29 +1100,34 @@
 fi
 fi
 
-AC_PATH_PROG(PATH_TO_NSLOOKUP,nslookup)
 
 AC_MSG_CHECKING(for nslookup syntax)
-if test -n "$PATH_TO_NSLOOKUP"
-then
-	if $PATH_TO_NSLOOKUP -sil 127.0.0.1 2>&1 | grep ["Invalid option: sil"] >/dev/null
+AC_ARG_WITH(nslookup_command,--with-nslookup-command=<syntax> sets syntax for nslookup,ac_cv_nslookup_command=$withval)
+if test -n "$$ac_cv_nslookup_command"; then
+	AC_MSG_RESULT([(command line) $ac_cv_nslookup_command])
+else
+	AC_PATH_PROG(PATH_TO_NSLOOKUP,nslookup)
+	if test -n "$PATH_TO_NSLOOKUP"
 	then
-		ac_cv_nslookup_command="$PATH_TO_NSLOOKUP"
-		AC_MSG_RESULT([$ac_cv_nslookup_command])
+		if $PATH_TO_NSLOOKUP -sil 127.0.0.1 2>&1 | grep ["Invalid option: sil"] >/dev/null
+		then
+			ac_cv_nslookup_command="$PATH_TO_NSLOOKUP"
+			AC_MSG_RESULT([$ac_cv_nslookup_command])
 
-	else
-		ac_cv_nslookup_command="$PATH_TO_NSLOOKUP -sil"
-		AC_MSG_RESULT([$ac_cv_nslookup_command])
+		else
+			ac_cv_nslookup_command="$PATH_TO_NSLOOKUP -sil"
+			AC_MSG_RESULT([$ac_cv_nslookup_command])
 
+		fi
+	else
+		AC_MSG_WARN([nslookup command not found])
 	fi
-	EXTRAS="$EXTRAS check_dns"
-
-else
-	AC_MSG_WARN([nslookup command not found])
 fi
 
-AC_DEFINE_UNQUOTED(NSLOOKUP_COMMAND,"$ac_cv_nslookup_command",
-	[path and args for nslookup])
+if test -n "$ac_cv_nslookup_command"; then
+	EXTRAS="$EXTRAS check_dns"
+	AC_DEFINE_UNQUOTED(NSLOOKUP_COMMAND,"$ac_cv_nslookup_command", [path and args for nslookup])
+fi
 
 AC_PATH_PROG(PATH_TO_HOST,host)
 if test -n "$ac_cv_path_PATH_TO_HOST"
--- ./configure.in~	2005-02-08 16:25:52.000000000 +0200
+++ ./configure.in	2005-03-27 23:09:00.000000000 +0300
@@ -1120,15 +1136,18 @@
 fi
 
 AC_PATH_PROG(PATH_TO_UPTIME,uptime)
+AC_ARG_WITH(uptime_command,--with-uptime-command=<path> sets path to uptime,PATH_TO_UPTIME=$withval)
 AC_DEFINE_UNQUOTED(PATH_TO_UPTIME,"$PATH_TO_UPTIME",[path to uptime binary])
 
 AC_PATH_PROG(PATH_TO_RPCINFO,rpcinfo)
+AC_ARG_WITH(rpcinfo_command,--with-rpcinfo-command=<path> sets path to rpcinfo,PATH_TO_RPCINFO=$withval)
 AC_DEFINE_UNQUOTED(PATH_TO_RPCINFO,"$PATH_TO_RPCINFO",[path to rpcinfo binary])
 
 AC_PATH_PROG(PATH_TO_NTPDATE,ntpdate)
+AC_ARG_WITH(ntpdate_command,--with-ntpdate-command=<path> sets path to ntpdate,PATH_TO_NTPDATE=$withval)
 AC_PATH_PROGS(PATH_TO_NTPDC,ntpdc xntpdc)
 AC_PATH_PROGS(PATH_TO_NTPQ,ntpq)
-if (test -x "$PATH_TO_NTPDATE" || test -x "$PATH_TO_NTPQ")
+if (test -n "$PATH_TO_NTPDATE" || test -n "$PATH_TO_NTPQ")
 then
 	AC_DEFINE_UNQUOTED(PATH_TO_NTPQ,"$PATH_TO_NTPQ",[path to ntpq binary])
 	AC_DEFINE_UNQUOTED(PATH_TO_NTPDATE,"$PATH_TO_NTPDATE",[path to ntpdate binary])
@@ -1145,7 +1164,8 @@
 fi
 
 AC_PATH_PROG(PATH_TO_SMBCLIENT,smbclient)
-if test -x "$PATH_TO_SMBCLIENT"
+AC_ARG_WITH(smbclient_command,--with-smbclient-command=<path> sets path to smbclient,PATH_TO_SMBCLIENT=$withval)
+if test -n "$PATH_TO_SMBCLIENT"
 then
 	AC_DEFINE_UNQUOTED(PATH_TO_SMBCLIENT,"$PATH_TO_SMBCLIENT",[path to smbclient binary])
 else
@@ -1221,8 +1241,8 @@
 fi
 
 AC_PATH_PROG(PATH_TO_SSH,ssh)
-
-if test -x "$PATH_TO_SSH"
+AC_ARG_WITH(ssh_command,--with-ssh-command=<path > sets path for ssh,PATH_TO_SSH=$withval)
+if test -n "$PATH_TO_SSH"
 then
 	AC_DEFINE_UNQUOTED(SSH_COMMAND,"$PATH_TO_SSH",[path to ssh binary])
 	EXTRAS="$EXTRAS check_by_ssh"
@@ -1232,7 +1252,8 @@
 
 
 AC_PATH_PROG(PATH_TO_MAILQ,mailq)
-if test -x "$PATH_TO_MAILQ"
+AC_ARG_WITH(mailq_command,--with-mailq-command=<path> sets path to mailq,PATH_TO_MAILQ=$withval)
+if test -n "$PATH_TO_MAILQ"
 then
 	AC_DEFINE_UNQUOTED(PATH_TO_MAILQ,"$PATH_TO_MAILQ",[path to mailq])
 else
--- ./configure.in~	2005-03-27 23:12:00.000000000 +0300
+++ ./configure.in	2005-04-09 18:29:55.000000000 +0300
@@ -1468,9 +1468,10 @@
 fi
 
 AC_PATH_PROG(PATH_TO_DIG,dig)
-AC_DEFINE_UNQUOTED(PATH_TO_DIG,"$PATH_TO_DIG",[Path to dig command, if present])
+AC_ARG_WITH(dig_command,--with-dig-command=<path> Path to dig command,PATH_TO_DIG=$withval)
 if test -n "$PATH_TO_DIG"; then
 	EXTRAS="$EXTRAS check_dig"
+	AC_DEFINE_UNQUOTED(PATH_TO_DIG,"$PATH_TO_DIG",[Path to dig command, if present])
 fi
 
 if test -f plugins/check_nt.c ; then
