--- nagios-plugins-1.4.15.orig/configure.in	2012-11-23 19:30:28.000000000 +0100
+++ nagios-plugins-1.4.15/configure.in	2012-11-23 19:53:42.847216408 +0100
@@ -254,12 +254,6 @@
 
 dnl Check for radius libraries
 _SAVEDLIBS="$LIBS"
-AC_CHECK_LIB(radiusclient,rc_read_config)
-if test "$ac_cv_lib_radiusclient_rc_read_config" = "yes"; then
-  EXTRAS="$EXTRAS check_radius"
-	RADIUSLIBS="-lradiusclient"
-  AC_SUBST(RADIUSLIBS)
-else
   AC_CHECK_LIB(radiusclient-ng,rc_read_config)
   if test "$ac_cv_lib_radiusclient_ng_rc_read_config" = "yes"; then
     EXTRAS="$EXTRAS check_radius"
@@ -269,7 +263,6 @@
     AC_MSG_WARN([Skipping radius plugin])
     AC_MSG_WARN([install radius libs to compile this plugin (see REQUIREMENTS).])
   fi
-fi
 LIBS="$_SAVEDLIBS"
 
 dnl Check for LDAP libraries
