--- contrib/check_snmp_procs.pl.orig	2008-04-23 12:13:17.833685899 +0000
+++ contrib/check_snmp_procs.pl	2008-04-23 12:15:18.456846167 +0000
@@ -59,6 +59,7 @@
 my $opt_version;
 my $opt_wanted_procs;
 my $opt_warning;
+my $opt_password;
 
 #
 my $max_no_processes = 999999;
@@ -123,6 +124,8 @@
 	exit $ERRORS{'UNKNOWN'};
 }
 
+my ($longest_wanted, $longest_current);
+
 ($longest_wanted, @wanted_procs) = parse_wanted_procs($opt_verbose, $opt_wanted_procs, $opt_warning, $opt_critical);
 
 $SIG{'ALRM'} = sub {
@@ -240,7 +243,7 @@
 
 	my($verbose, $host, $username, $privpassword, $authprotocol, $authpassword, $community, $port, $oidname, $snmp_version) = @_;
 	my(%process_list, %process_pid_list, $result);
-	my $process_list_longest = 1, $not_done = 1;
+	my ($process_list_longest, $not_done) = (1, 1);
 	my(@args, @oids, $oid, $name);
 
 	($session, $error) = Net::SNMP->session(
