--- ./plugins/check_ping.c~~	2005-05-02 19:41:24.000000000 +0300
+++ ./plugins/check_ping.c	2005-05-02 19:43:39.000000000 +0300
@@ -401,6 +401,7 @@
 {
 	char buf[MAX_INPUT_BUFFER];
 	int result = STATE_UNKNOWN;
+	char *oldlocale;
 
 	if ((child_process = spopen (cmd)) == NULL)
 		die (STATE_UNKNOWN, _("Could not open pipe: %s\n"), cmd);
@@ -409,6 +410,11 @@
 	if (child_stderr == NULL)
 		printf (_("Cannot open stderr for %s\n"), cmd);
 
+	// force LC_NUMERIC to "C" otherwise scanning %f from ping program will not
+	// "see" decimal separator if locale decimal separator is comma and my ping
+	// program knows nothing of locales (well it's having always decimal
+	// separator)
+	oldlocale = setlocale (LC_NUMERIC, "C");
 	while (fgets (buf, MAX_INPUT_BUFFER - 1, child_process)) {
 
 		result = max_state (result, error_scan (buf, addr));
@@ -434,6 +440,7 @@
 				 sscanf(buf,"rtt min/avg/max/mdev = %*f/%f/%*f/%*f ms",&rta)==1)
 			continue;
 	}
+	setlocale (LC_NUMERIC, oldlocale);
 
 	/* this is needed because there is no rta if all packets are lost */
 	if (pl == 100)
