--- ./m4/fsusage.m4~	2004-12-15 22:51:47.000000000 +0200
+++ ./m4/fsusage.m4	2005-03-28 20:38:43.000000000 +0300
@@ -239,7 +239,7 @@
 # Prerequisites of lib/fsusage.c not done by jm_FILE_SYSTEM_USAGE.
 AC_DEFUN([gl_PREREQ_FSUSAGE_EXTRA],
 [
-  AC_REQUIRE([jm_AC_TYPE_UINTMAX_T])
+  AC_REQUIRE([gl_AC_TYPE_UINTMAX_T])
   AC_CHECK_HEADERS_ONCE(fcntl.h)
   AC_CHECK_HEADERS(dustat.h sys/fs/s5param.h sys/filsys.h sys/statfs.h sys/statvfs.h)
   jm_STATFS_TRUNCATES
