diff -ur nagios-plugins-1.4.3.orig/configure.in nagios-plugins-1.4.3/configure.in
--- nagios-plugins-1.4.3.orig/configure.in	2006-04-19 05:42:33.000000000 +0200
+++ nagios-plugins-1.4.3/configure.in	2006-12-21 13:32:51.000000000 +0100
@@ -1639,7 +1639,8 @@
 		#endif],
 		[va_list args;],
 		[AC_MSG_RESULT(yes)],
-		[NEED_VA_LIST=-DNEED_VA_LIST AC_SUBST(NEED_VA_LIST) AC_MSG_RESULT(no)])
+		[NEED_VA_LIST=-DNEED_VA_LIST
+		AC_SUBST(NEED_VA_LIST) AC_MSG_RESULT(no)])
 
 case $host in
 	*bsd*)
