--- nagios-plugins-1.4.13/plugins-scripts/check_flexlm.pl	2003-02-04 08:16:16.000000000 +0200
+++ nagios-plugins-1.4.13-pp/plugins-scripts/check_flexlm.pl	2009-02-16 12:28:04.993531068 +0200
@@ -17,7 +17,7 @@
 #host[anchor]=any host will do;some.address.com;;check-host-alive;3;120;24x7;1;1;1;
 #service[anchor]=yodel;24x7;3;5;5;unix-admin;60;24x7;1;1;1;;check_flexlm!/opt/lic/licfiles/yodel_lic
 #service[anchor]=yeehaw;24x7;3;5;5;unix-admin;60;24x7;1;1;1;;check_flexlm!/opt/lic/licfiles/yeehaw_lic
-#command[check_flexlm]=/some/path/libexec/check_flexlm.pl $ARG1$
+#command[check_flexlm]=/usr/lib/nagios/plugins/check_flexlm.pl $ARG1$
 #
 # Notes:
 # - you need the lmstat utility which comes with flexlm.
--- nagios-plugins-1.4.13/plugins-scripts/check_rpc.pl	2003-08-10 15:11:49.000000000 +0300
+++ nagios-plugins-1.4.13-pp/plugins-scripts/check_rpc.pl	2009-02-16 12:28:05.023530932 +0200
@@ -10,7 +10,7 @@
 #
 # Use these hosts.cfg entries as examples
 #
-# command[check_nfs]=/some/path/libexec/check_rpc $HOSTADDRESS$ nfs
+# command[check_nfs]=/usr/lib/nagios/plugins/check_rpc $HOSTADDRESS$ nfs
 # service[check_nfs]=NFS;24x7;3;5;5;unix-admin;60;24x7;1;1;1;;check_rpc
 #
 # initial version: 3 May 2000 by Truongchinh Nguyen and Karl DeBisschop
