--- nagios-plugins-2.0/plugins/check_hpjd.c~	2014-04-21 11:57:30.000000000 +0300
+++ nagios-plugins-2.0/plugins/check_hpjd.c	2014-04-21 12:56:26.818058154 +0300
@@ -41,7 +41,7 @@
 #define DEFAULT_COMMUNITY "public"
 #define DEFAULT_PORT "161"
 
-const char *option_summary = "-H host [-C community]\n";
+const char *option_summary = "-H host [-C community] [-n]\n";
 
 #define HPJD_LINE_STATUS           ".1.3.6.1.4.1.11.2.3.9.1.1.2.1"
 #define HPJD_PAPER_STATUS          ".1.3.6.1.4.1.11.2.3.9.1.1.2.2"
@@ -67,6 +67,7 @@
 char *community = NULL;
 char *address = NULL;
 char *port = NULL;
+int nopaperout = 0;
 
 int
 main (int argc, char **argv)
@@ -243,7 +244,9 @@
 			strcpy (errmsg, _("Paper Jam"));
 		}
 		else if (paper_out) {
-			result = STATE_WARNING;
+			if (!nopaperout) {
+				result = STATE_WARNING;
+			}
 			strcpy (errmsg, _("Out of Paper"));
 		}
 		else if (line_status == OFFLINE) {
@@ -317,6 +318,7 @@
 /*  		{"critical",       required_argument,0,'c'}, */
 /*  		{"warning",        required_argument,0,'w'}, */
   		{"port", required_argument,0,'p'}, 
+		{"nopaperout", no_argument, 0, 'n'},
 		{"version", no_argument, 0, 'V'},
 		{"help", no_argument, 0, 'h'},
 		{0, 0, 0, 0}
@@ -327,7 +329,7 @@
 
 
 	while (1) {
-		c = getopt_long (argc, argv, "+hVH:C:p:", longopts, &option);
+		c = getopt_long (argc, argv, "+hnVH:C:p:", longopts, &option);
 
 		if (c == -1 || c == EOF || c == 1)
 			break;
@@ -345,6 +349,9 @@
 		case 'C':									/* community */
 			community = strscpy (community, optarg);
 			break;
+		case 'n':									/* nopaperout */
+			nopaperout = 1;
+			break;
 		case 'V':									/* version */
 			print_revision (progname, revision);
 			exit (STATE_OK);
@@ -404,7 +411,9 @@
 
 	printf (" %s\n", "-C, --community=STRING");
 	printf ("    %s", _("The SNMP community name "));
+	printf (" %s\n", "-n, --nopaperout");
 	printf (_("(default=%s)"), DEFAULT_COMMUNITY);
+	printf ("    %s\n", "No return WARNING state on \"Out of paper\"");
 	printf ("\n");
 
 #ifdef NP_EXTRA_OPTS
@@ -437,5 +439,5 @@
 print_usage (void)
 {
   printf ("%s\n", _("Usage:"));
-	printf ("%s -H host [-C community] [-p port]\n", progname);
+	printf ("%s -H host [-C community] [-p port] [-n]\n", progname);
 }
