--- contrib/check_snmp_disk_monitor.pl  2008-04-23 14:27:47.390299906 +0000
+++ contrib/check_snmp_disk_monitor.pl~ 2008-04-23 13:27:18.000000000 +0000
@@ -7,7 +7,7 @@

 use strict;
 require 5.6.0;
-use lib qw( /opt/nagios/libexec );
+use lib qw( /usr/lib/nagios/plugins );
 use utils qw(%ERRORS $TIMEOUT &print_revision &support &usage);
 use SNMP 5.0;
 use Getopt::Long;
--- contrib/check_snmp_procs.pl.orig    2008-04-23 12:13:17.833685899 +0000
+++ contrib/check_snmp_procs.pl 2008-04-23 12:15:18.456846167 +0000
@@ -32,10 +32,10 @@
 #    make it work with snmp version 3
 #    Suggestions???
 #
-#use strict;
+use strict;
 use Getopt::Long;
 use Net::SNMP qw (oid_lex_sort oid_base_match SNMP_VERSION_1);
-use lib "/usr/local/nagios/libexec";
+use lib "/usr/lib/nagios/plugins";
 use utils qw(%ERRORS &print_revision &support &usage);

 my $PROGNAME="check_snmp_procs";

